app/code/Bulbulatory/Recommendations/Controller/Adminhtml/Listing/Delete.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Bulbulatory\Recommendations\Controller\Adminhtml\Listing;
+
+/**
+ * Class Delete
+ * @package Bulbulatory\Recommendations\Controller\Adminhtml\Listing
+ */
+class Delete extends AbstractDeleteAction
+{
+    public function execute()
+    {
+        $id = (int)$this->_request->getParam('id');
+        $resultRedirect = $this->resultRedirectFactory->create();
+
+        if (!empty($id) && $this->deleteRecommendationsRowsByIds([$id]) > 0) {
+            $this->messageManager->addSuccessMessage(__('Recommendation with ID: ' . $id . ' deleted!'));
+        } else {
+            $this->messageManager->addErrorMessage(__('Invalid request'));
```

Comment:
Dobrze byłoby zwrócić informację o tym, dlaczego invalid. Wystarczy zwykłe `missing id parameter`, ale gdy wystąpią jakieś problemy to już będzie to jakiś punkt zaczepienia.